### PR TITLE
Marshalling the value in set command

### PR DESCRIPTION
**Pull request by lovish2525**
Issue link - https://github.com/DiceDB/dice/issues/750

Changes Made:

1. Handling of values to ensure proper serialization and deserialization of nested map structures in SET command.
2. Added UTs

### CODE REVIEW

File: internal/server/utils/redisCmdAdapter_test.go [hunk 1]
```diff
@@ -26,6 +26,14 @@ func TestParseHTTPRequest(t *testing.T) {
 			expectedCmd:  "SET",
 			expectedArgs: []string{"k1", "v1", "nx"},
 		},
+		{
```
Comment (pratikpandey21) [inline internal/server/utils/redisCmdAdapter_test.go:29]: Can we add more tests for this?
Have arrays and submaps inside value

Comment (lovish2525) [reply internal/server/utils/redisCmdAdapter_test.go:29]: Added

**Review by pratikpandey21 — APPROVED** [review]
Please fix the tests, rest looks good.

**Comment by lovish2525** [comment]
@pratikpandey21 Just want to confirm if the escape characters in API response is fine?
"{\"foo\":{\"bar\":1}}"

Screenshot 2024-09-29 at 11 03 41 PM

**Comment by lovish2525** [comment]
@pratikpandey21 Testcases are running fine, Please help in next steps

**Comment by lucifercr07** [comment]
> @pratikpandey21 Just want to confirm if the escape characters in API response is fine?
> 
> Screenshot 2024-09-29 at 11 03 41 PM

I don't think escape chars should be there in response. It should be a proper json. @lovish2525 please check once.
cc: @pratikpandey21

**Comment by vinitparekh17** [comment]
> > @pratikpandey21 Just want to confirm if the escape characters in API response is fine?
> > Screenshot 2024-09-29 at 11 03 41 PM
> 
> I don't think escape chars should be there in response. It should be a proper json. @lovish2525 please check once. cc: @pratikpandey21

In redis JSON.GET i have noticed escape characters there should not be problem with that as long as client is able to parse it without having errors

**Comment by lovish2525** [comment]
@vinitparekh17 @pratikpandey21 Fixed http UTs

**Comment by lovish2525** [comment]
@vinitparekh17 So are we good to go with escape chars?

**Comment by lucifercr07** [comment]
@lovish2525 looks like redis does have escape chars in response. I believe we can keep the same.
cc: @pratikpandey21

**Comment by lovish2525** [comment]
@lucifercr07 Kindly approve the workflow

**Comment by vinitparekh17** [comment]
@lovish2525 lgtm
@lucifercr07 hold on this marge for a while, i am about to create pr for http response in proper json format

**Pull request merged**